R/tiq-test.R

```diff
@@ -0,0 +1,606 @@
+#'        the enriched entities ("enriched")
+#' @param select.sources a chararacter vector of the sources on the dataset you want
+#'        to be a part of the test, or NULL for all of them
+#' @return a numeric matrix with the overlap ratios
+#' @export
+overlapTest <- function(group, date, type="raw", select.sources=NULL) {
+
+  test_that("tiq.test.overlapTest: parameters must have correct types", {
+    expect_that(class(group), equals("character"))
+    expect_match(date, "^[0123456789]{8}$", info="must be a date (YYYYMMDD)")
+    expect_that(class(type), equals("character"))
+  })
+
+  # Loading the data from the specific date
+  ti.dt = loadTI(type, group, date=date)
+  split.ti = split(ti.dt, ti.dt$source)
+
+  ## Performing the overlap test only on the sources we select
+  if (is.null(select.sources)) {
+    select.sources = unique(ti.dt$source)
+  }
+
+  overlap.matrix = matrix(nrow=length(select.sources), ncol=length(select.sources),
+                          dimnames=list(select.sources, select.sources))
+
+  # Removing duplicated entries from each source
+  for (ti in 1:length(select.sources)) {
+    setkey(split.ti[[select.sources[ti]]], entity)
+    split.ti[[select.sources[ti]]] = unique(split.ti[[select.sources[ti]]])
+  }
+
+  # Calculating the overlap for each pairing
+  for (ti in 1:length(select.sources)) {
+    for (overlap in 1:length(select.sources)) {
+      # For each pairing
+      overlap.count = tiq.helper.overlapCount(split.ti[[select.sources[ti]]],
+                                              split.ti[[select.sources[overlap]]])
+      overlap.matrix[ti,overlap] = overlap.count /
+                                   length(unique(split.ti[[select.sources[ti]]]$entity))
+    }
+  }
+
+  return(overlap.matrix)
+}
+
+#' Plots the results of the overlap test in a (mostly) clear heatmap for comparisons
+#'
+#' @param overlap the output of the 'tiq.test.OverlapTest' function
+#' @param title a title for your plot
+#' @param plot.sources a chararacter vector of the sources on the novelty test you want
+#'        to be a part of the plot, or NULL for all of them
+#' @return ggplot2 object (plots when printed)
+#' @export
+plotOverlapTest <- function(overlap, title="Overlap Test Plot", plot.sources=NULL) {
+
+  if (!is.matrix(overlap) || (dim(overlap)[1] != dim(overlap)[2])) {
+    msg = sprintf("tiq.test.plotOverlapTest: 'overlap' parameter mush be a square matrix")
+    flog.error(msg)
+    stop(msg)
+  }
+
+  plot.data = as.data.table(melt(overlap))
+
+  if (!is.null(plot.sources)) {
+    plot.data = plot.data[as.character(Var1) %chin% plot.sources]
+    plot.data = plot.data[as.character(Var2) %chin% plot.sources]
+  }
+
+  gg <- ggplot(plot.data, aes(x=Var1, y=Var2))
+  gg <- gg + geom_tile(aes(fill=value), color="#e3e3e3", size=0.5)
+  gg <- gg + coord_equal()
+  gg <- gg + scale_fill_distiller(palette="YlOrBr", name="%\nOverlap", labels=percent)
```

Comment:
Variable `percent` does not exist here in this context. My "fix" was to edit it out the `labels=percent` part, as i have no idea what it does. :)